src/server.c

```diff
@@ -3276,16 +3274,24 @@ static void propagatePendingCommands() {
  * What we want to achieve is that the entire execution unit will be done atomically,
  * currently with respect to replication and post jobs, but in the future there might
  * be other considerations. So we basically want the `postUnitOperations` to trigger
- * after the entire chain finished.
- *
- * Current, in order to avoid massive code changes that could be risky to cherry-pick,
- * we count on the mechanism we already have such as `server.core_propagation`,
- * `server.module_ctx_nesting`, and `server.in_nested_call`. We understand that we probably
- * do not need all of those variable and we will make an attempt to re-arrange it on unstable
- * branch. */
+ * after the entire chain finished. */
 void postExecutionUnitOperations() {
+    if (server.execution_nesting)
+        return;
+
     firePostExecutionUnitJobs();
+
+    /* If we are at the top-most call() and not inside a an active module
+     * context (e.g. within a module timer) we can propagate what we accumulated. */
     propagatePendingCommands();
+
+    if (server.busy_module_yield_flags) {
```

Comment:
Why did you move this code here? Was it a bug? Do we have tests for it?

Reply:
I moved it because it always appeared after postExecutionUnitOperations in module.c, so I thought it was an excellent opportunity to save code duplication

on the other hand, it means that now this code will be executed even when running vanilla commands, but I think it doesn't have any effect, in that case, WDYT? (i can't remember - did you write the RM_Yield PR?)

Reply:
I did not write it (https://github.com/redis/redis/pull/9963).
I do not see that today it is appeared after each time we call `postExecutionUnitOperations` (what am I missing?).

Reply:
not every time, just in module.c

@perryitay do you see any danger in running
```
if (server.busy_module_yield_flags) {
        blockingOperationEnds();
        server.busy_module_yield_flags = BUSY_MODULE_YIELD_NONE;
        if (server.current_client)
            unprotectClient(server.current_client);
        unblockPostponedClients();
    }
```
after running a non-module command?

Reply:
IIUC, `server.busy_module_yield_flags` should be 0 when running vanilla commands

Reply:
@perryitay ping ^

Reply:
i think its better to ask @oranagra

Reply:
i don't think moving it will cause any damage (that flag is set when we're inside a module call stack, and unset before we leave), so other execution units can't intervene at that time anyway.
i.e. we won't have any eviction, expiration or anything else before the module exists.

however, i think it is very confusing to see it in postExecutionUnitOperations (in server.c), and i'd rather this logic be kept when we destroy a module context.
if we wanna share code, we can create a dedicated function, and call it when we destroy the context, let's not move it to a general place where that code is out of it's context.

Reply:
after giving it some thought i decided that having `modulePostExecutionUnitOperations` is the cleanest solution (otherwise, in the case of a module command, `postExecutionUnitOperations` will be called twice: once from `moduleFreeContext` and once in `afterCommand`